### ISSUE TITLE
Nested try blocks reference wrong exception object

### ISSUE THREAD

**Issue by amitmurthy** [JuliaLang/julia#5032]
```
try 
    error("Foo")
catch 
    # Print additional debugging information
    # Ignore any errors while doing so 
    try
        error("Bar")
    catch 
    end
    rethrow()
end
```

throws `Bar` and not `Foo`. 

While 

```
try 
    error("Foo")
catch e
    try
        error("Bar")
    catch 
    end
    rethrow(e)
end
```

works correctly.

**Comment by felixcremer** [comment]
This was fixed in #28878 for version 1.1 and later.
Should this be backported to 1.0.x?

**Comment by StefanKarpinski** [comment]
Exception stacks are a major feature and definitely not eligible for backport. I suspect this is unfixable on 1.0.x since you'd need to introduce something similar there which is not allowed.

**Comment by Keno** [comment]
Closing as fixed on master by #28878.
